core/qbft/qbft.go

```diff
@@ -202,6 +203,29 @@ func Run[I any, V comparable](ctx context.Context, d Definition[I, V], t Transpo
 		dedupIn = dedup
 	}
 
+	isDuplicatedRule := func(rule uponRule) bool {
+		if rule == uponJustifiedDecided {
```

Comment:
On second thought, I think it is **_sending_** of `qCommit` that may be triggered multiple times per round, not the **_receiving_** of `qCommit`.  Receiving it once is sufficient, since then we have decided and we are done. So I think we must remove this line.

Reply:
Done